tests/ui/duplicate_pymodule_submodule.stderr

```diff
@@ -4,8 +4,14 @@ error: `submodule` may only be specified once (it is implicitly always specified
 4 |     mod submod {}
   |     ^^^
 
-error[E0433]: failed to resolve: use of undeclared crate or module `submod`
- --> tests/ui/duplicate_pymodule_submodule.rs:4:6
+error[E0425]: cannot find value `_PYO3_DEF` in module `submod`
```

Comment:
This error looks a bit strange; is the result of the macro failing early with errors and not emitting the complete submodule.

I think changes like #4243 can improve this in future.